src/engines_gpl/dflowfm/packages/dflowfm_kernel/src/dflowfm_data/unstruc_inifields.f90

```diff
@@ -655,7 +655,7 @@ subroutine readIniFieldProvider(inifilename, node_ptr, groupname, quantity, file
          end if
 
          ! read value
-         if (filetype == inside_polygon) then
+         if ((filetype == inside_polygon) .and. .not. strcmpi(quantity, 'initialvertical', 15)) then
```

Comment:
Let's try and not introduce hardcoded checks on quantity names anymore, unless absolutely unavoidable.
In this case, there's a more elegant alternative:
```suggestion
         if (filetype == inside_polygon .and. method == METHOD_CONSTANT) then
```